contracts/access/TimelockController.sol

```diff
@@ -0,0 +1,294 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.6.0;
+pragma experimental ABIEncoderV2;
+
+import "./../math/SafeMath.sol";
+import "./AccessControl.sol";
+
+/**
+ * @dev Contract module which acts as a timelocked controller. When set as the
+ * owner of an `Ownable` smart contract, it enforces a timelock on all
+ * `onlyOwner` maintenance operations. This gives time for users of the
+ * controlled contract to exit before a potentially dangerous maintenance
+ * operation is applied.
+ *
+ *
+ * This contract is designed to be self administered, meaning it should be its
+ * own administrator. The proposer (resp executor) role is in charge of
+ * proposing (resp executing) operation. A common use case is to position this
+ * {TimelockController} as the owner of a smart contract, with a multisig or a
+ * DAO as the sole proposer. Once at least one proposer and one executor have
+ * been appointed, self-administration can be enable using the `makeLive`
+ * function.
+ */
+contract TimelockController is AccessControl {
```

Comment:
By convention all of our non-view functions should be `virtual`. (See [How we plan to adopt Solidity 0.6 virtual functions](https://forum.openzeppelin.com/t/how-we-plan-to-adopt-solidity-0-6-virtual-functions/2063).)